zebra-chain/src/primitives/address.rs

```diff
@@ -119,4 +119,27 @@ impl Address {
     pub fn is_transparent(&self) -> bool {
         matches!(self, Self::Transparent(_))
     }
+
+    /// Returns the payment address for transparent or sapling addresses.
+    ///
+    /// # Panics
+    ///
+    /// - If the address is of sapling type and no `zcash_address::Network` is provided.
+    #[allow(clippy::unwrap_in_result)]
+    pub fn payment_address(&self, network: Option<zcash_address::Network>) -> Option<String> {
+        use zcash_address::{ToAddress, ZcashAddress};
+
+        match &self {
+            Self::Transparent(address) => Some(address.to_string()),
+            Self::Sapling { address, .. } => {
+                let data = address.to_bytes();
+                let address = ZcashAddress::from_sapling(
+                    network.expect("For a sapling payment address, network should be provided."),
+                    data,
+                );
+                Some(address.encode())
+            }
```

Comment:
The network parameter is the same as `Sapling { network }` in the current code. In future code, it could cause consensus-critical bugs if we allow the caller to supply a different network.

```suggestion
            Self::Sapling { address, network } => {
                let data = address.to_bytes();
                let address = ZcashAddress::from_sapling(
                    network.into(),
                    data,
                );
                Some(address.encode())
            }
```